tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,556 @@
+        })?;
+        let uri = unsafe { std::ffi::CStr::from_ptr(c_uri) };
+        Ok(String::from(uri.to_string_lossy()))
+    }
+
+    pub fn query_type(
+        &self,
+        context: &'ctx Context,
+    ) -> TileDBResult<QueryType> {
+        let c_context = self.context.capi();
+        let mut c_type: ffi::tiledb_query_type_t = out_ptr!();
+        context.capi_return(unsafe {
+            ffi::tiledb_group_get_query_type(
+                c_context,
+                Self::capi(self),
+                &mut c_type,
+            )
+        })?;
+        QueryType::try_from(c_type)
+    }
+
+    pub fn delete_group<S>(
+        &self,
+        context: &'ctx Context,
+        uri: S,
+        recursive: bool,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let ctx = context.capi();
+        let c_uri = cstring!(uri.as_ref());
+        let c_recursive: u8 = if recursive { 1 } else { 0 };
+        context.capi_return(unsafe {
+            ffi::tiledb_group_delete_group(
+                ctx,
+                Self::capi(self),
+                c_uri.as_ptr(),
+                c_recursive,
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn add_member<S, T>(
+        &self,
+        context: &'ctx Context,
+        uri: S,
+        relative: bool,
+        name: Option<T>,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+        T: AsRef<str>,
+    {
+        let ctx = context.capi();
+        let c_uri = cstring!(uri.as_ref());
+        let c_ptr = match name {
+            None => std::ptr::null::<std::ffi::c_char>(),
+            Some(s) => cstring!(s.as_ref()).as_ptr(),
+        };
+        let c_relative: u8 = if relative { 1 } else { 0 };
+        context.capi_return(unsafe {
+            ffi::tiledb_group_add_member(
+                ctx,
+                Self::capi(self),
+                c_uri.as_ptr(),
+                c_relative,
+                c_ptr,
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn delete_member<S>(
+        &self,
+        context: &'ctx Context,
+        name_or_uri: S,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let ctx = context.capi();
+        let c_name_or_uri = cstring!(name_or_uri.as_ref());
+        context.capi_return(unsafe {
+            ffi::tiledb_group_remove_member(
+                ctx,
+                Self::capi(self),
+                c_name_or_uri.as_ptr(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn get_member_count(
+        &self,
+        context: &'ctx Context,
+    ) -> TileDBResult<u64> {
+        let ctx = context.capi();
+        let mut c_count: u64 = out_ptr!();
+        context.capi_return(unsafe {
+            ffi::tiledb_group_get_member_count(
+                ctx,
+                Self::capi(self),
+                &mut c_count,
+            )
+        })?;
+        Ok(c_count)
+    }
+
+    pub fn get_member_by_index(
```

Comment:
On `main` we now have a `DimensionKey` which will soon be re-named `LookupKey` by the fragment info PR. Using that, you can/should consolidate `get_member_by_index` and `get_member_by_name` into a single function `member`